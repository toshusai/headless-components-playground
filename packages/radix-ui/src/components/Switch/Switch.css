.switch {
  all: unset;
  min-width: 48px;
  height: 24px;
  background-color: lightgray;
  border-radius: 9999px;
  position: relative;
}

.switch:focus-visible {
  box-shadow: 0 0 0 2px cadetblue;
}

.switch[data-state="checked"] {
  background-color: lightblue;
}

.switch-thumb {
  display: block;
  background-color: white;
  width: 20px;
  height: 20px;
  border-radius: 9999px;
  transition: transform 100ms;
  transform: translateX(2px);
  will-change: transform;
}

.switch-thumb[data-state="checked"] {
  transform: translateX(26px);
}

.label {
  display: flex;
  align-items: center;
  gap: 8px;
}
