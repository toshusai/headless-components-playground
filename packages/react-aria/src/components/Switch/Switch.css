.react-aria-Switch {
  display: flex;
  gap: 8px;
  align-items: center;

  .indicator {
    width: 48px;
    height: 24px;
    background-color: lightgray;
    border-radius: 9999px;
    position: relative;
    align-items: center;
    display: flex;

    &:before {
      content: "";
      display: block;
      background-color: white;
      width: 20px;
      height: 20px;
      border-radius: 9999px;
      transition: transform 100ms;
      transform: translateX(2px);
      will-change: transform;
    }
  }

  &[data-pressed] .indicator {
    background-color: lightblue;

    &:before {
      transform: translateX(26px);
    }
  }

  &[data-selected] {
    .indicator {
      background-color: lightblue;

      &:before {
        transform: translateX(26px);
      }
    }
  }

  &[data-focus-visible] .indicator {
    box-shadow: 0 0 0 2px cadetblue;
  }
}
