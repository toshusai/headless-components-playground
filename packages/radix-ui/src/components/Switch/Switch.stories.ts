import type { Meta, StoryObj } from "@storybook/react";

import { Switch } from "./Switch";

const meta = {
  title: "RadixUI/Switch",
  component: Switch,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Switch>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    label: "Switch",
  },
};
