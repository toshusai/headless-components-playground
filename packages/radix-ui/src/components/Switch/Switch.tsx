import * as RadixSwitch from "@radix-ui/react-switch";
import "./Switch.css";

export const Switch = (props: { label: string }) => (
  <label className="label">
    <RadixSwitch.Root className="switch">
      <RadixSwitch.Thumb className="switch-thumb" />
    </RadixSwitch.Root>
    {props.label}
  </label>
);
