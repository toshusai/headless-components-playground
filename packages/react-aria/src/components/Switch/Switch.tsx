import { Switch as RacSwitch } from "react-aria-components";
import "./Switch.css";

export function Switch(props: { label: string }) {
  return (
    <RacSwitch>
      <div className="indicator" />
      {props.label}
    </RacSwitch>
  );
}
