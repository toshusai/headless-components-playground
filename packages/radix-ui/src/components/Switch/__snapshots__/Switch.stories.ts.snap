// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadixUI/Switch Primary smoke-test 1`] = `
<label class="label">
  <button type="button"
          role="switch"
          aria-checked="false"
          data-state="unchecked"
          value="on"
          class="switch"
  >
    <span data-state="unchecked"
          class="switch-thumb"
    >
    </span>
  </button>
  Switch
</label>
`;
